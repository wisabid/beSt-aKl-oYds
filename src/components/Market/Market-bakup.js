import React, { useEffect } from 'react';
import Outcome from '../Outcome'

const Market = (props) => {
    const { mdata, eventid, outcomedata, showOutcomOndemand, showMarketOndemand } = props;

    useEffect(() => {
        console.log(' C  o m ppppppp! DID Mount! Shabz');
    }, [])

    return (
        <ul >
            {
                mdata.map((market) => {
                    if (market.data.eventId === eventid) {
                        return (
                            <>
                            <li><b>(M)</b><span onClick={() => showMarketOndemand(market.data.marketId)}>{market.data.marketId} - {market.data.name}</span> <br />
                            - {market.data.type}   <br />                                
                            <Outcome odata={outcomedata} eventid={eventid} marketid={market.data.marketId} 
                                showOutcomOndemand={(id) => showOutcomOndemand(id)}
                                mdata={mdata}   
                            />     

                            </li>                               
                            </>
                        )
                    }
                })
            }
            
        </ul>
    )
}

export default Market;