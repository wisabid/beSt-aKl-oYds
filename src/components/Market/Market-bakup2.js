import React, { useEffect } from 'react';
import Outcome from '../Outcome'

const Market = (props) => {
    const { mdata, eventid, outcomedata, showOutcomOndemand, showMarketOndemand, marketdata } = props;

    useEffect(() => {
        console.log(' C  o m ppppppp! DID Mount! Shabz');
        props.showMarketOndemand(mdata);
    }, [])
    console.log(marketdata)
    if (marketdata.length) {
        return (
            <ul >
                {
                    marketdata.map((market) => {
                        if (market.data.eventId === eventid) {
                            return (
                                <li>
                                    <b>(M)</b><span onClick={() => showMarketOndemand(market.data.marketId)}>{market.data.marketId} - {market.data.name}</span>
                                </li>
                            )
                        }
                    })
                }
                
            </ul>
        )
    }
    else {
        return (
            <h4>Loading...</h4>
        )
    }
}

export default Market;