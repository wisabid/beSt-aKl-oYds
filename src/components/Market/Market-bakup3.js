import React, { useEffect, useReducer } from 'react';
import Outcome from '../../containers/Outcome';
import Subscription from '../../containers/Subscription';
import soccer from '../../assets/images/Soccerball.svg';

function marketreducer(state, action) {
    switch(action.type) {
        case 'mtype':
            return {...state};
            break;
        default : 
            return {...state}
    }
}

const Market = (props) => {
    
    const { mdata=[], eventid, outcomedata, showOutcomOndemand, showMarketOndemand, pmarket, ondmdata, subscriptions, markets=[], showmarkets, marketdata=[] } = props;
    const [ state, dispatch ] = useReducer(marketreducer, []);
    useEffect(() => {
        if (mdata.length === 0 && markets.length) {
           showmarkets(markets);            
        }
    }, []);
    if (marketdata.length && !pmarket) {
        //mdata = [...marketdata];
    }
    if (mdata.length || marketdata.length) {
        if (pmarket) {        
                return (
                    <fieldset>
                        <legend>Primary Market</legend>
                        <ul className="noborder">
                            <li key={mdata[0].marketId}>
                                <div className="inrow">
                                    <span className="anchorl" onClick={() => showMarketOndemand(mdata[0].marketId)}>{mdata[0].name} ( <span onClick={() => dispatch({type: 'mtype', payload: mdata[0].type})}>{mdata[0].type}</span> )</span> <br />
                                    <span className="justtitle" title="Live Price Limit">{mdata[0].liabilities.livePriceLimit} ( LPL )</span>
                                    <Subscription 
                                        uid={`m.${mdata[0].marketId}`}
                                    />   
                                </div>                                                              
                                <Outcome eventid={eventid} marketid={mdata[0].marketId} 
                                    showOutcomOndemand={(id) => showOutcomOndemand(id)}
                                    mdata={mdata[0]}   
                                />     

                            </li> 
                        </ul>  
                    </fieldset>
                )
        
        }
        else {
            return (
                <fieldset>
                    <legend>Market(s)</legend>
                <ul className="noborder">
                    {
                        marketdata.map((market, indx) => {
                            if (market.eventId === eventid) {
                                return (
                                    <>
                                    <li key={market.marketId}>
                                        <div className="inrow">
                                            <span className="anchorl" onClick={() => showMarketOndemand(market.marketId)}><b>{indx+1}. </b>{market.name} ( {market.type} )</span> 
                                            <span className="justtitle" title="Live Price Limit">{market.liabilities.livePriceLimit} ( LPL )</span>
                                            <Subscription 
                                                uid={`m.${market.marketId}`}
                                            />   
                                        </div>
                                        {market.name
                                            ?<Outcome eventid={eventid} marketid={market.marketId} 
                                            showOutcomOndemand={(id) => showOutcomOndemand(id)}
                                            mdata={mdata}   
                                        />   
                                            :null                                            
                                        }                          
                                          

                                    </li>                               
                                    </>
                                )
                            }
                        })

                    }
                    {
                        props.ondemandmarketdata.map((market) => {
                            
                                return (
                                    <>
                                    <li key={market.marketId}>
                                    <div className="inrow">
                                        <span className="anchorl" onClick={() => props.showMarketOndemandAlt(market.marketId)}>{market.name?`${market.name} ( ${market.type} )`:`Market ( ${market.marketId} )`}</span>                             
                                        <span className="justtitle" title="Live Price Limit">{market.liabilities?`${market.liabilities.livePriceLimit} ( LPL ) `:null} </span>
                                        <Subscription 
                                            uid={`m.${market.marketId}`}
                                        />
                                    </div>
                                        
                                     {market.name
                                        ?<Outcome eventid={eventid} marketid={market.marketId} 
                                        showOutcomOndemand={(id) => showOutcomOndemand(id)}
                                        mdata={mdata} 
                                        ondemand="true"  
                                    />
                                        :null

                                     }   
                                     <hr />
                                        

                                    </li>                               
                                    </>
                                )
                        })
                    }
                    
                </ul>
                </fieldset>
            )
        }
    }
    else {
        return (
            <img src={soccer} className="bao-spinner" alt="logo" />
        )
    }
}

export default Market;