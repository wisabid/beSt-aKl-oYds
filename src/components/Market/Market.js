import React from 'react';
import Outcome from '../Outcome';
import Subscription from '../../containers/Subscription';

const Market = (props) => {
    const { mdata, eventid, outcomedata, showOutcomOndemand, showMarketOndemand, pmarket, ondmdata, subscriptions } = props;
    
    if (pmarket) {
        if (mdata.length) {
            return (
                <fieldset>
                <legend>Primary Market</legend>
                <ul className="noborder">
                <li><b>(M)</b><span onClick={() => showMarketOndemand(mdata[0].marketId)}>{mdata[0].marketId} - {mdata[0].name}</span> <br />
                - {mdata[0].type}   
                <Subscription 
                    uid={`m.${mdata[0].marketId}`}
                />   
                <br />                                
                <Outcome odata={outcomedata} eventid={eventid} marketid={mdata[0].marketId} 
                    showOutcomOndemand={(id) => showOutcomOndemand(id)}
                    mdata={mdata[0]}   
                />     

                </li> 
                </ul>  
                </fieldset>
            )
        }
        else {
            return (
                <ul>
                    <li>Suspended</li>
                </ul>
            )
        }
    }
    else {
        return (
            <fieldset>
                <legend>Market(s)</legend>
            <ul className="noborder">
                {
                    mdata.map((market) => {
                        if (market.eventId === eventid) {
                            return (
                                <>
                                <li><b>(M)</b><span onClick={() => showMarketOndemand(market.marketId)}>{market.marketId} - {market.name}</span> <br />
                                - {market.type}   
                                <Subscription 
                                    uid={`m.${market.marketId}`}
                                />   
                                <br />                                
                                <Outcome odata={outcomedata} eventid={eventid} marketid={market.marketId} 
                                    showOutcomOndemand={(id) => showOutcomOndemand(id)}
                                    mdata={mdata}   
                                />     

                                </li>                               
                                </>
                            )
                        }
                    })

                }
                {
                    props.ondemandmarketdata.map((market) => {
                        
                            return (
                                <>
                                <li><b>(M)</b><span onClick={() => props.showMarketOndemandAlt(market.marketId)}>{market.marketId} - EXTRAAAAA</span> <br />                               
                                - {market.type}   
                                <Subscription 
                                    uid={`m.${market.marketId}`}
                                />
                                <br />                                
                                <Outcome odata={outcomedata} eventid={eventid} marketid={market.marketId} 
                                    showOutcomOndemand={(id) => showOutcomOndemand(id)}
                                    mdata={mdata}   
                                />    

                                </li>                               
                                </>
                            )
                    })
                }
                
            </ul>
            </fieldset>
        )
    }
}

export default Market;