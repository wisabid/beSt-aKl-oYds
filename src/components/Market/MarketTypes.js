import React from 'react';
import Subscription from '../../containers/Subscription';
import Outcome from '../../containers/Outcome';

const MarketTypes = (props) => {
    const { marketdata } = props;
    debugger;
    let newmarketdata = marketdata.filter(item => props.dta[0].indexOf(item.marketId) !== -1)
    return (
        <>
            <button onClick={props.goback} className="backbtn anchorl"> Home </button>
            <div className="bao-live">            
                <fieldset>
                    <legend>Category Based Market(s) Display</legend>
                    <ul className="noborder">
                        {
                            newmarketdata.map((market, indx) => {
                                    return (
                                        <>
                                        <li key={market.marketId}>
                                            <div className="inrow">
                                                <span className="anchorl"><b>{indx+1}. </b>{market.name}</span> 
                                                <span className="anchorl">( {market.type} )</span>
                                                <span className="justtitle" title="Live Price Limit">{market.liabilities.livePriceLimit} ( LPL )</span>
                                                <Subscription 
                                                    uid={`m.${market.marketId}`}
                                                />   
                                            </div>
                                            {market.name
                                                ?<Outcome eventid={market.eventId} marketid={market.marketId} />   
                                                :null                                            
                                            }                          
                                            

                                        </li>                               
                                        </>
                                    )  
                            })

                        }          
                        
                    </ul>
                </fieldset>
            </div>
        </>
    )
}

export default MarketTypes;