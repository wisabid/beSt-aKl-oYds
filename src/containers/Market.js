import { connect } from 'react-redux';
import MarketComp from '../components/Market';
import { outcomeOnDemand, showMarketOndemand, showMarketOndemandAlt } from '../store/actions'
import displayWrapper from '../utils/displayWrapper';

const mapStateToProps = (state) => {
    return {        
        outcomedata : state.rL.outcomedata,
        marketdata : state.rL.marketdata,
        ondemandmarketdata : state.rL.ondemandmarketdata
    }
}

const mapdispatchToProps = (dispatch) => {
    return {
        showMarketOndemand : (id) => dispatch(showMarketOndemand([id])),
        showMarketOndemandAlt : (id) => dispatch(showMarketOndemandAlt([id])),
        showOutcomOndemand : (id) => dispatch(outcomeOnDemand([id], true))
    }
}

const Market = connect(mapStateToProps, mapdispatchToProps)(displayWrapper(MarketComp));

export default Market;