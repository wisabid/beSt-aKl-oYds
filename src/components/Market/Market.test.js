import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Market from './Market';

jest.mock('../../context/UserContext');

describe('<Counter />', () => {
  //const mdata = JSON.parse('[{"type":"MARKET_DATA","data":{"marketId":93649849,"eventId":21249939,"name":"Both Score No Draw","displayOrder":-32496,"type":"standard","status":{"active":true,"resulted":false,"cashoutable":false,"displayable":true,"suspended":false,"noExtraTime":false,"live":true},"liabilities":{"livePriceLimit":2500},"spAvail":false,"outcomes":[367530493,367530501]}},{"type":"MARKET_DATA","data":{"marketId":93649179,"eventId":21249939,"name":"Both Teams To Score","displayOrder":-32499,"type":"standard","status":{"active":true,"resulted":false,"cashoutable":true,"displayable":true,"suspended":false,"noExtraTime":false,"live":true},"liabilities":{"livePriceLimit":2500},"spAvail":false,"outcomes":[367528211,367528215]}},{"type":"MARKET_DATA","data":{"marketId":93649150,"eventId":21249939,"name":"Away Team Clean Sheet","displayOrder":-32018,"type":"standard","status":{"active":true,"resulted":false,"cashoutable":false,"displayable":true,"suspended":false,"noExtraTime":false,"live":true},"liabilities":{"livePriceLimit":1000},"spAvail":false,"outcomes":[367528121,367528125]}},{"type":"MARKET_DATA","data":{"marketId":93649398,"eventId":21249939,"name":"Home Team Clean Sheet","displayOrder":-32019,"type":"standard","status":{"active":true,"resulted":false,"cashoutable":false,"displayable":true,"suspended":true,"noExtraTime":false,"live":true},"liabilities":{"livePriceLimit":1000},"spAvail":false,"outcomes":[367528987,367528997]}},{"type":"MARKET_DATA","data":{"marketId":93649188,"eventId":21249939,"name":"Correct Score","displayOrder":-32498,"type":"correct-score","status":{"active":true,"resulted":false,"cashoutable":true,"displayable":true,"suspended":true,"noExtraTime":false,"live":true},"liabilities":{"livePriceLimit":5000},"spAvail":false,"outcomes":[367528229,367528236,367528242,367528247,367528256,367528260,367528267,367528271,367528277,367528281,367528288,367528296,367528301,367528305,367528309,367528314,367528319,367528322,367528327,367528331,367528334,367528340,367528344,367528349,367528355,368052632,368052633,368078338,368078339,368078340,368078341]}},{"type":"MARKET_DATA","data":{"marketId":93649223,"eventId":21249939,"name":"Double Chance","displayOrder":-32497,"type":"standard","status":{"active":true,"resulted":false,"cashoutable":true,"displayable":true,"suspended":false,"noExtraTime":false,"live":true},"liabilities":{"livePriceLimit":2500},"spAvail":false,"outcomes":[367528381,367528386,367528390]}},{"type":"MARKET_DATA","data":{"marketId":93649220,"eventId":21249939,"name":"Draw No Bet","displayOrder":-32550,"type":"standard","status":{"active":true,"resulted":false,"cashoutable":false,"displayable":true,"suspended":false,"noExtraTime":false,"live":true},"liabilities":{"livePriceLimit":2000},"spAvail":false,"outcomes":[367528365,367528370]}},{"type":"MARKET_DATA","data":{"marketId":93649155,"eventId":21249939,"name":"Away Team Exact Number of Goals","displayOrder":-32015,"type":"standard","status":{"active":true,"resulted":false,"cashoutable":false,"displayable":true,"suspended":true,"noExtraTime":false,"live":true},"liabilities":{"livePriceLimit":2000},"spAvail":false,"outcomes":[367528136,367528140,367528143,367528149,367528156,367528160,367528163,368052623]}},{"type":"MARKET_DATA","data":{"marketId":93649228,"eventId":21249939,"name":"Exact Number Of Goals","displayOrder":-32017,"type":"standard","status":{"active":true,"resulted":false,"cashoutable":false,"displayable":true,"suspended":false,"noExtraTime":false,"live":true},"liabilities":{"livePriceLimit":2000},"spAvail":false,"outcomes":[367528401,367528406,367528411,367528416,367528420,367528425,367528429,367528436,367528442,368052634,368078342]}},{"type":"MARKET_DATA","data":{"marketId":93649407,"eventId":21249939,"name":"Home Team Exact Number of Goals","displayOrder":-32016,"type":"standard","status":{"active":true,"resulted":false,"cashoutable":false,"displayable":true,"suspended":false,"noExtraTime":false,"live":true},"liabilities":{"livePriceLimit":1000},"spAvail":false,"outcomes":[367529014,367529020,367529026,367529037,367529042,367529050,367529060]}}]');

  const MarketComp = shallow(<Market mdata={[]} pmarket="true"/>)

  it('<Market /> renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Market />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('<Market /> component matches the snapshot', () => {
    expect(MarketComp).toMatchSnapshot();
  })

  it('<Market /> component displays the primary market', () => {
    //expect(MarketComp.find('.primary-market')).toHaveLength(1);
  })

});
