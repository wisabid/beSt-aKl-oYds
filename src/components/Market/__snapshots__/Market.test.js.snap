// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> <Market /> component matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Market
    mdata={Array []}
    pmarket="true"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "alt": "logo",
      "className": "bao-spinner",
      "src": "Soccerball.svg",
    },
    "ref": null,
    "rendered": null,
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "logo",
        "className": "bao-spinner",
        "src": "Soccerball.svg",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
